services/src/main/scala/cromwell/services/metadata/impl/WriteMetadataActor.scala

```diff
@@ -55,18 +62,28 @@ class WriteMetadataActor(batchSize: Int, flushRate: FiniteDuration)
     case Event(FlushBatchToDb, HasData(e)) =>
       log.debug("Flushing {} metadata events to the DB", e.length)
       // blech
-      val events = e.toVector.collect({ case e: MetadataEvent => e })
-      addMetadataEvents(events) onComplete {
-        case Success(_) => self ! DbWriteComplete
+      val events: Vector[MetadataEvent] = e.toVector.collect({ case e: MetadataEvent => e})
+      val eventsToAcknowledge: Map[Iterable[MetadataEvent], ActorRef] = e.toVector.collect({ case PutMetadataActionAndRespond(e, replyTo) => e -> replyTo }) toMap
+      val allEvents: Iterable[MetadataEvent] = eventsToAcknowledge.keys.foldLeft(events)((acc, ev) => {acc ++ ev })
+      addMetadataEvents(allEvents) onComplete {
+        case Success(_) =>
+          self ! DbWriteComplete
+          if(eventsToAcknowledge.nonEmpty) {
```

Comment:
The reply's are only required when there are put events that require a response. @mcovarr you mean do something like a foreach instead of checking to see if there are any put actions that require acknowledgement?